### PR TITLE
Add kubernetes_service option to teleport helm chart

### PR DESCRIPTION
**Pull request by stevenGravy**
Allows for using kubernetes_service option instead of kubernetes enabled under proxy.  This allows for pointing at multiple other k8s clusters with a kubeconfig.

### CODE REVIEW

File: examples/chart/teleport/templates/config.yaml [hunk 1]
```diff
@@ -93,17 +93,27 @@ data:
...
         listen_addr: {{ .Values.config.listen_addr }}:{{ .Values.ports.proxykube.containerPort }}
+
+
+{{- else if .Values.config.teleport.kubernetes_service }}
+      #kube_service declaration
+      kube_listen_addr: {{ .Values.config.listen_addr }}:{{ .Values.ports.proxykube.containerPort }}
+    kubernetes_service:
+{{ toYaml .Values.config.teleport.kubernetes_service  | indent 8 }}
+{{- end }}
 {{- end }}
```
Comment (awly) [inline examples/chart/teleport/templates/config.yaml:115]: User should be able to set both `.Values.config.teleport.proxy_service.kubernetes` and `.Values.config.teleport.kubernetes_service`:
```suggestion
{{- else if .Values.config.teleport.kubernetes_service }}
      #kube_service declaration
      kube_listen_addr: {{ .Values.config.listen_addr }}:{{ .Values.ports.proxykube.containerPort }}
{{- end }}

{{- if .Values.config.teleport.kubernetes_service }}
    kubernetes_service:
{{ toYaml .Values.config.teleport.kubernetes_service  | indent 6 }}
{{- end }}
```

File: examples/chart/teleport/values.yaml [hunk 2]
```diff
@@ -160,6 +160,14 @@ config:
         # Specify a different hostname for the k8s public address (if different to config.public_address)
         # public_addr: teleportkubernetes.example.com
         
+    # To use a kubernetes_service remove kubernetes under proxy_service
```
Comment (awly) [inline examples/chart/teleport/values.yaml:163]:
```suggestion
    # To use a kubernetes_service uncomment this section        
```

File: examples/chart/teleport/values.yaml [hunk 3]
```diff
@@ -160,6 +160,14 @@ config:
         # Specify a different hostname for the k8s public address (if different to config.public_address)
         # public_addr: teleportkubernetes.example.com
         
+    # To use a kubernetes_service remove kubernetes under proxy_service        
+    # kubernetes_service:
+      # enabled: true            
+      # labels:
+      #   type: staging
```
Comment (awly) [inline examples/chart/teleport/values.yaml:167]:
```suggestion
      # labels:
      #   example-label: example-value
```

**Review by awly — COMMENTED** [review]
@stevenGravy could you please test this with:
- `proxy_service` enabled, `kubernetes_service` not enabled
- `proxy_service` not enabled, `kubernetes_service` enabled
- both `proxy_service` and `kubernetes_service` enabled
- both `proxy_service` and `kubernetes_service` enabled and `proxy_service.kubernetes_public_addr` set

**Review by klizhentas — APPROVED** [review]

**Comment by stevenGravy** [comment]
@awly , I have confirmed:

**Comment by stevenGravy** [comment]
I have confirmed.

**Comment by stevenGravy** [comment]
@awly  I confirmed all of the scenarios.  Also reconfirmed with a optional `kubeconfig_file` 
**confirmed**: `proxy_service` enabled, `kubernetes_service` not enabled
**confirmed**: proxy_service not enabled, kubernetes_service enabled
** confirmed**: both proxy_service and kubernetes_service enabled
** confirmed**: both proxy_service and kubernetes_service enabled and proxy_service.kubernetes_public_addr set

**Review by awly — APPROVED** [review]

**Pull request merged**